lms/djangoapps/course_api/api.py

```diff
@@ -57,7 +62,20 @@ def course_detail(request, username, course_key):
     )
 
 
-def list_courses(request, username, org=None, filter_=None):
+def _filter_courses_by_role(course_queryset, user, access_type):
+    """
+    Return a course queryset filtered by the access type for which the user has access.
+    """
+    return LazySequence(
+        (
+            course for course in course_queryset
+            if has_access(user, access_type, course.id)
+        ),
+        est_len=len(course_queryset)
```

Comment:
Actually, this `est_len` is needed for correct results. The overriding of `__len__()` in `LazySequence` prevents the QuerySet from being evaluated until the response is evaluated at the end of the client request.

Reply:
Yeah, the `course_api` was the poster-child for switching over to bridgekeeper, but we need to do a bit more conversion of `has_access` first, so that we can use it to modify the queryset, rather than checking the permissions afterwards.